app/models/like.rb

```diff
@@ -1,4 +1,11 @@
 class Like < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  belongs_to :user
   belongs_to :post
+  after_create :update_likes_counter
+  after_destroy :update_likes_counter
+  after_save :update_post_counter
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```

Comment:
```suggestion
  private
  
  def update_likes_counter
    post.update(likes_counter: post.likes.count)
  end
```

[OPTIONAL]
- [ ] Same for the `update_likes_counter`, kindly make it private